src/NzbDrone.Common/Http/HttpRequest.cs

```diff
@@ -54,6 +54,7 @@ public Uri Url
         public Dictionary<string, string> Cookies { get; private set; }
         public bool StoreResponseCookie { get; set; }
         public TimeSpan RateLimit { get; set; }
+        public HttpRequestProxySettings Proxy {get; set;}
```

Comment:
The crux of why this is an issue, is because IConfigService is not available in the Common module

If we could expose IConfigService to the Common module, then we could just inject it into HttpClient (which could then pass it around to the dispatchers).

This would remove the need for any special interface just for proxy settings, and would remove the interceptor. 

It should just  be a matter of refactoring IConfigServe to the common module.